### DISCUSSION TITLE
Use of local secrets

### DISCUSSION THREAD

**Discussion by sti0** [PrefectHQ/prefect#4060] [Q&A] [answered]
Hi there,

I have a question about the use of local secrets placed in the config.toml file. Is it right, that if I register the flow to another instance the local secrets form my config.toml are attached to the flow? And if I execute the flow it runs with the secret values of my config.toml instead of the values of the other instance? And even if I use it fully local, the flow needs to re-registered after changing the secrets?

I ask because the [docs ](https://docs.prefect.io/core/concepts/secrets.html#overview) are not clear to me in that point:

> Very often, workflows require sensitive information to run: API keys, passwords, tokens, credentials, etc. As a matter of best practice, such information **should never be hardcoded into a workflow's source code**, as the code itself will need to be guarded. 

> Secret tasks, which are special tasks that can be used in your flow when working with sensitive information. Unlike regular tasks, Secret tasks are designed to access sensitive information at runtime and use a special SecretResult to ensure the **results are not stored.**

Thanks + BR

**Comment by zanieb** [comment, marked as answer]
Hi @sti0 

> Is it right, that if I register the flow to another instance the local secrets form my config.toml are attached to the flow?

This is not the case. That instance will need the secrets stored in its local config. If you are using Prefect Cloud, you can store them in a secure vault and add secrets via the UI/API--then they will be available everywhere.

> And even if I use it fully local, the flow needs to re-registered after changing the secrets?

This is also not the case. The secrets are pulled by the flow at runtime.

**Reply by sti0** [reply]
@madkinsz thank you for your answer. It was my fault. I used the Secret as described [here](https://docs.prefect.io/core/concepts/secrets.html#local-context). If I use the [PrefectSecret](https://docs.prefect.io/api/latest/tasks/secrets.html#prefectsecret) it works as expected.
